content/docs/contributing/release-process.md

```diff
@@ -476,62 +489,36 @@ page if a step is missing or if it is outdated.
        [`cert-manager/website`](https://github.com/cert-manager/website) in
        order to:
 
-       - Update the section "Supported releases" in the
-         [supported-releases](../installation/supported-releases.md) page.
        - Update the section "Supported releases" in the
          [supported-releases](../installation/supported-releases.md) page.
        - Update the section "How we determine supported Kubernetes versions" on
          the [supported-releases](../installation/supported-releases.md) page.
          In the table, set "n/a" for the line where "next periodic" is since
          these tests will be disabled until we do our first alpha.
 
-    7. **(final release only)** Open a PR to
-       [`jetstack/testing`](https://github.com/jetstack/testing) and change Prow's
-       config. To do this, take inspiration from [Maartje's PR
-       example](https://github.com/jetstack/testing/pull/397/files).
-
-    8. **(final release only)** Push a new release branch to
-       [`cert-manager/cert-manager`](https://github.com/cert-manager/cert-manager). If the
-       final release is `v1.0.0`, then push the new branch `release-1.1`:
-
-        ```bash
-        # Must be run from the cert-manager repo folder.
-        git checkout -b release-1.1 v1.0.0
-        git push origin release-1.1
-        ```
-
-    9. **(final release only)** Open a PR to
-       [`cert-manager/website`](https://github.com/cert-manager/website) with
-       updates to the website configuration. To do this, take inspiration from
-       [Maartje's PR
-       example](https://github.com/cert-manager/website/pull/309/files).
-
-    10. Ensure that any installation commands in
+    7. Ensure that any installation commands in
        [`cert-manager/website`](https://github.com/cert-manager/website) install
        the latest version. This should be done after every release, including
        patch releases as we want to encourage users to always install the latest
-       patch.
-
-    11. Future: check that our Algolia search indexing is up-to-date for the website - i.e. that the new version of the docs
-       is being indexed correctly. This is listed here as it's a step we should be checking after a release of a major version
-       but at the time of writing we don't know how to do it!
+       patch. In addition, ensure that release notes for the latest version are added.
 
-    12. Open a PR against the Krew index such as [this one](https://github.com/kubernetes-sigs/krew-index/pull/1724),
-        bumping the versions of our kubectl plugins.
+    8. Open a PR against the Krew index such as [this one](https://github.com/kubernetes-sigs/krew-index/pull/1724),
+      bumping the versions of our kubectl plugins. This is likely only worthwhile if
+      cmctl / kubectl plugin functionality has changed significantly or after the first release of a new major version.
```

Comment:
note: I don't see any reason to do this for a standard patch release most of the time, so I added a note to that effect.